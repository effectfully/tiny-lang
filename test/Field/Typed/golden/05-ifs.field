let field'  : field  = if T then 0 else 0;
let bool'   : bool   = if T then T else T;
let vector' : vector = if T then { T } else { T };
