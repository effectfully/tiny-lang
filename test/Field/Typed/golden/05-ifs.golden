Program (Statements [ELet (UniVar {_uniVarUni = Field, _uniVarVar = field'_3}) (EIf (EConst (UniConst Bool True)) (EConst 0) (EConst 0)),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?bool_4}) (EIf (EConst (UniConst Bool True)) (EConst (UniConst Bool True)) (EConst (UniConst Bool True))),ELet (UniVar {_uniVarUni = Vector, _uniVarVar = #vector_5}) (EIf (EConst (UniConst Bool True)) (EConst (UniConst Vector [True])) (EConst (UniConst Vector [True])))])